.github/workflows/update_app_templates_hash.yml

```diff
@@ -0,0 +1,45 @@
+name: Update aws/aws-sam-cli with latest commit hash from aws/aws-sam-cli-app-templates
+
+on:
+  schedule:
+    - cron: "0 * * * *" # run at the top of every hour
+  workflow_dispatch: {}
+
+jobs:
+  updateCommitHash:
+    permissions:
+      pull-requests: write
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout App Templates
+        uses: actions/checkout@v3
+        with:
+          repository: aws/aws-sam-cli-app-templates
+          path: aws-sam-cli-app-templates
+
+      - name: Checkout SAM CLI
+        uses: actions/checkout@v3
+        with:
+          repository: aws/aws-sam-cli
+          path: aws-sam-cli
+
+      - name: Update hash & Raise PR for SAM CLI
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
nit (or tip for the future): it can help to reduce attack surface (in practice I don't see any real issue here) by separating the steps, such that _only_ the steps that require `GITHUB_TOKEN` are in their own step, and all others are in another without `GITHUB_TOKEN`

Reply:
Good suggestion. AFAIK each step runs in separate shell that is why I need to combine everything into single step. Otherwise we need to pass these variables between steps. I will give it another try to limit the scope of this token, if so will update the PR with that.

Reply:
Applied your suggestion and separated step into 2 to limit the token's visibility